there are many warning when building bootloader for an stm32f4

``` 
/mnt/c/workspace/tinyuf2/ports/stm32f4/board_flash.c: In function 'flash_write':
/mnt/c/workspace/tinyuf2/ports/stm32f4/board_flash.c:158:71: warning: cast increases required alignment of target type [-Wcast-align]
  158 |   if (HAL_FLASH_Program(FLASH_TYPEPROGRAM_WORD, dst + i, (uint64_t) (*(uint32_t*)(src + i)) ) != HAL_OK) {
      |                                                                       ^
/mnt/c/workspace/tinyuf2/lib/st/stm32f4xx_hal_driver/Src/stm32f4xx_hal_flash_ex.c: In function 'FLASH_MassErase':
/mnt/c/workspace/tinyuf2/lib/st/stm32f4xx_hal_driver/Src/stm32f4xx_hal_flash_ex.c:953:60: warning: unused parameter 'Banks' [-Wunused-parameter]
  953 | static void FLASH_MassErase(uint8_t VoltageRange, uint32_t Banks)
      |                                                   ~~~~~~~~~^~~~~
/mnt/c/workspace/tinyuf2/lib/st/stm32f4xx_hal_driver/Src/stm32f4xx_hal_flash_ex.c: In function 'FLASH_OB_EnableWRP':/mnt/c/workspace/tinyuf2/lib/st/stm32f4xx_hal_driver/Src/stm32f4xx_hal_flash_ex.c:1032:74: warning: unused parameter 'Banks' [-Wunused-parameter]
 1032 | static HAL_StatusTypeDef FLASH_OB_EnableWRP(uint32_t WRPSector, uint32_t Banks)
      |                                                                 ~~~~~~~~~^~~~~
/mnt/c/workspace/tinyuf2/lib/st/stm32f4xx_hal_driver/Src/stm32f4xx_hal_flash_ex.c: In function 'FLASH_OB_DisableWRP':
/mnt/c/workspace/tinyuf2/lib/st/stm32f4xx_hal_driver/Src/stm32f4xx_hal_flash_ex.c:1068:75: warning: unused parameter 'Banks' [-Wunused-parameter]
 1068 | static HAL_StatusTypeDef FLASH_OB_DisableWRP(uint32_t WRPSector, uint32_t Banks)
      |                                                                  ~~~~~~~~~^~~~~
/mnt/c/workspace/tinyuf2/lib/st/stm32f4xx_hal_driver/Src/stm32f4xx_hal_uart.c: In function 'HAL_UART_Transmit':
/mnt/c/workspace/tinyuf2/lib/st/stm32f4xx_hal_driver/Src/stm32f4xx_hal_uart.c:1061:15: warning: cast increases required alignment of target type [-Wcast-align]
 1061 |         tmp = (uint16_t *) pData;
      |               ^
/mnt/c/workspace/tinyuf2/lib/st/stm32f4xx_hal_driver/Src/stm32f4xx_hal_uart.c: In function 'HAL_UART_Receive':
/mnt/c/workspace/tinyuf2/lib/st/stm32f4xx_hal_driver/Src/stm32f4xx_hal_uart.c:1148:15: warning: cast increases required alignment of target type [-Wcast-align]
 1148 |         tmp = (uint16_t *) pData;
      |               ^
/mnt/c/workspace/tinyuf2/lib/st/stm32f4xx_hal_driver/Src/stm32f4xx_hal_uart.c: In function 'UART_Transmit_IT':
/mnt/c/workspace/tinyuf2/lib/st/stm32f4xx_hal_driver/Src/stm32f4xx_hal_uart.c:2927:13: warning: cast increases required alignment of target type [-Wcast-align]
 2927 |       tmp = (uint16_t *) huart->pTxBuffPtr;
      |             ^
/mnt/c/workspace/tinyuf2/lib/st/stm32f4xx_hal_driver/Src/stm32f4xx_hal_uart.c: In function 'UART_Receive_IT':
/mnt/c/workspace/tinyuf2/lib/st/stm32f4xx_hal_driver/Src/stm32f4xx_hal_uart.c:2999:13: warning: cast increases required alignment of target type [-Wcast-align]
 2999 |       tmp = (uint16_t *) huart->pRxBuffPtr;
      |             ^
LINK _build/circuitart_stm32f4core/tinyuf2-circuitart_stm32f4core.elf
CREATE _build/circuitart_stm32f4core/tinyuf2-circuitart_stm32f4core.bin
CREATE _build/circuitart_stm32f4core/tinyuf2-circuitart_stm32f4core.hex
``` 
![stm32f4 build warnings](https://user-images.githubusercontent.com/85825139/121896407-d71e5100-cd53-11eb-840f-9eeb78eaa67f.PNG)

--- a/ports/stm32f3/board_flash.c
+++ b/ports/stm32f3/board_flash.c
@@ -112,7 +112,8 @@ void flash_write(uint32_t dst, const uint8_t *src, int len)
 
   for (int i = 0; i < len; i += 4)
   {
-    HAL_FLASH_Program(FLASH_TYPEPROGRAM_WORD, dst + i, (uint64_t) (*(uint32_t*)(src + i)) );
+    uint32_t data = *( (uint32_t*) ((void*) (src + i)) );
+    HAL_FLASH_Program(FLASH_TYPEPROGRAM_WORD, dst + i, (uint64_t) data);
   }
 
   if (memcmp((void*)dst, src, len) != 0)

--- a/ports/stm32f4/board_flash.c
+++ b/ports/stm32f4/board_flash.c
@@ -155,14 +155,18 @@ void flash_write(uint32_t dst, const uint8_t *src, int len)
 
 	for (int i = 0; i < len; i += 4)
 	{
-		if (HAL_FLASH_Program(FLASH_TYPEPROGRAM_WORD, dst + i, (uint64_t) (*(uint32_t*)(src + i)) ) != HAL_OK) {
-			TU_LOG1("Failed to write flash at address %08lX", dst + i);
-			break;
-		};
-		if (FLASH_WaitForLastOperation(HAL_MAX_DELAY) != HAL_OK) {
-			TU_LOG1("Waiting on last operation failed");
-			return;
-		};
+	  uint32_t data = *( (uint32_t*) ((void*) (src + i)) );
+    if ( HAL_FLASH_Program(FLASH_TYPEPROGRAM_WORD, dst + i, (uint64_t) data) != HAL_OK )
+    {
+      TU_LOG1("Failed to write flash at address %08lX", dst + i);
+      break;
+    }
+
+    if ( FLASH_WaitForLastOperation(HAL_MAX_DELAY) != HAL_OK )
+    {
+      TU_LOG1("Waiting on last operation failed");
+      return;
+    }
 	}
 
 	if (memcmp((void*)dst, src, len) != 0)

--- a/ports/stm32f4/board_flash.c
+++ b/ports/stm32f4/board_flash.c
@@ -91,31 +91,35 @@ static struct {
 
 static uint8_t erasedSectors[BOARD_FLASH_SECTORS];
 
-uint32_t flash_func_sector_size(unsigned sector)
+uint32_t flash_func_sector_size (unsigned sector)
 {
-	if (sector < BOARD_FLASH_SECTORS) {
-		return flash_sectors[sector].size;
-	}
+  if ( sector < BOARD_FLASH_SECTORS )
+  {
+    return flash_sectors[sector].size;
+  }
 
-	return 0;
+  return 0;
 }
 
-static bool is_blank(uint32_t addr, uint32_t size) {
-		for (uint32_t i = 0; i < size; i += sizeof(uint32_t)) {
-			if (*(uint32_t*)(addr + i) != 0xffffffff) {
-				return false;
-			}
-		}
-		return true;
+static bool is_blank (uint32_t addr, uint32_t size)
+{
+  for ( uint32_t i = 0; i < size; i += sizeof(uint32_t) )
+  {
+    if ( *(uint32_t*) (addr + i) != 0xffffffff )
+    {
+      return false;
+    }
+  }
+  return true;
 }
 
 void flash_write(uint32_t dst, const uint8_t *src, int len)
 {
-	// assume sector 0 (bootloader) is same size as sector 1
-	uint32_t addr = flash_func_sector_size(0) + (APP_LOAD_ADDRESS & 0xfff00000);
-	uint32_t sector = 0;
-	int erased = false;
-	uint32_t size = 0;
+  // assume sector 0 (bootloader) is same size as sector 1
+  uint32_t addr = flash_func_sector_size(0) + (APP_LOAD_ADDRESS & 0xfff00000);
+  uint32_t sector = 0;
+  int erased = false;
+  uint32_t size = 0;
 
   for ( unsigned i = 0; i < BOARD_FLASH_SECTORS; i++ )
   {
@@ -130,32 +134,33 @@ void flash_write(uint32_t dst, const uint8_t *src, int len)
     addr += size;
   }
 
-	if (sector == 0)
-	{
-	  TU_LOG1("invalid sector");
-	}
+  if ( sector == 0 )
+  {
+    TU_LOG1("invalid sector");
+  }
 
-	HAL_FLASH_Unlock();
+  HAL_FLASH_Unlock();
 
-	if (!erased && !is_blank(addr, size))
-	{
-		TU_LOG1("Erase: %08lX size = %lu\n", addr, size);
+  if ( !erased && !is_blank(addr, size) )
+  {
+    TU_LOG1("Erase: %08lX size = %lu\n", addr, size);
 
-		FLASH_Erase_Sector(sector, FLASH_VOLTAGE_RANGE_3);
-		if (FLASH_WaitForLastOperation(HAL_MAX_DELAY) != HAL_OK) {
-			TU_LOG1("Waiting on last operation failed");
-			return;
-		};
+    FLASH_Erase_Sector(sector, FLASH_VOLTAGE_RANGE_3);
+    if ( FLASH_WaitForLastOperation(HAL_MAX_DELAY) != HAL_OK )
+    {
+      TU_LOG1("Waiting on last operation failed");
+      return;
+    };
 
-		if (!is_blank(addr, size))
-		{
-		  TU_LOG1("failed to erase!");
-		}
-	}
+    if ( !is_blank(addr, size) )
+    {
+      TU_LOG1("failed to erase!");
+    }
+  }
 
-	for (int i = 0; i < len; i += 4)
-	{
-	  uint32_t data = *( (uint32_t*) ((void*) (src + i)) );
+  for ( int i = 0; i < len; i += 4 )
+  {
+    uint32_t data = *((uint32_t*) ((void*) (src + i)));
     if ( HAL_FLASH_Program(FLASH_TYPEPROGRAM_WORD, dst + i, (uint64_t) data) != HAL_OK )
     {
       TU_LOG1("Failed to write flash at address %08lX", dst + i);
@@ -167,12 +172,12 @@ void flash_write(uint32_t dst, const uint8_t *src, int len)
       TU_LOG1("Waiting on last operation failed");
       return;
     }
-	}
+  }
 
-	if (memcmp((void*)dst, src, len) != 0)
-	{
-	  TU_LOG1("failed to write");
-	}
+  if ( memcmp((void*) dst, src, len) != 0 )
+  {
+    TU_LOG1("failed to write");
+  }
 }
 
 //--------------------------------------------------------------------+

